model/metrics/gen-ai.yaml

```diff
@@ -0,0 +1,46 @@
+groups:
+  - id: metric_attributes.gen_ai
+    type: attribute_group
+    brief: 'This group describes GenAI metrics attributes'
+    attributes:
+      - ref: error.type
+        requirement_level:
+          conditionally_required: "if the operation ended in an error"
+        note: |
+          The cardinality of `error.type` SHOULD be low.
+
+          When working across multiple models, it is RECOMMENDED to use a common set of error types.
+
+          Additional details may be captured in domain-specific attributes.
+      - ref: server.address
+        requirement_level: recommended
+      - ref: server.port
+        requirement_level:
+          conditionally_required: If `sever.address` is set.
+      - ref: gen_ai.response.model
+        requirement_level: recommended
+      - ref: gen_ai.request.model
+        requirement_level: required
+      - ref: gen_ai.system
+        requirement_level: required
+      - ref: gen_ai.operation.name
+        requirement_level: required
+  - id: metric.gen_ai.client.token.usage
```

Comment:
We should separate this to input token usage and output token usage since the cost of them is different
`metric.gen_ai.client.usage.input_token`
`metric.gen_ai.client.usage.output_token`

Reply:
We had earlier conversations about this and chose to keep it as one instrument with an attribute `token_type` with values of `input` and `output` (and ability to add custom values). Effectively this gives you the totals, the subtotals, and ability to add custom token types in the future.

Should we revisit this conversation with new information?

Reply:
I'm fine having them in a single or a separate metrics. In both cases it's possible to derive totals or calculate costs. E.g. `token_cost_usd * gen_ai.client.token.usage{token.type="input"}` would give user a derived metric for prompt tokens, similarly to `token_cost_usd * gen_ai.client.token.usage.input_token`.

It could be problematic to add a new metrics if we're going to grow the list of token types.